mods/aero-tray.wh.cpp

```diff
@@ -0,0 +1,986 @@
+    LPARAM    lParam,
+    DWORD_PTR dwRefData
+)
+{
+    switch (uMsg)
+    {
+        case WM_PAINT:
+        {
+            PAINTSTRUCT ps;
+            HDC hDC = BeginPaint(hWnd, &ps);
+            LRESULT lr = PaintCustomizeLink(hWnd, hDC);
+            EndPaint(hWnd, &ps);
+            return lr;
+        }
+        case WM_PRINTCLIENT:
+        {
+            return PaintCustomizeLink(hWnd, (HDC)wParam);
+        }
+        /* Repaint and set cursor on hover */
+        case WM_MOUSEMOVE:
+        {
+            POINT pt;
+            pt.x = GET_X_LPARAM(lParam);
+            pt.y = GET_Y_LPARAM(lParam);
+
+            RECT rc;
+            GetClientRect(hWnd, &rc);
+
+            BOOL bHovered = g_bCustomizeHovered;
+            if (IsWindowEnabled(hWnd) && PtInRect(&rc, pt))
+            {
+                if (!g_bCustomizeCapturing && SetCapture(hWnd))
+                {
+                    g_bCustomizeCapturing = TRUE;
+                }
+                g_bCustomizeHovered = TRUE;
+            }
+            else
+            {
+                if (g_bCustomizeCapturing && ReleaseCapture())
+                {
+                    g_bCustomizeCapturing = FALSE;
+                }
+                g_bCustomizeHovered = FALSE;
+                g_bCustomizeDown = FALSE;
+            }
+
+            if (g_bCustomizeHovered != bHovered)
+            {
+                SetCursor(g_bCustomizeHovered ? g_hcHand : g_hcArrow);
+                InvalidateRect(hWnd, NULL, TRUE);
+            }
+            break;
+        }
+        /* Unhover when it's hidden */
+        case WM_SHOWWINDOW:
+            if (!wParam && g_bCustomizeHovered)
+            {
+                g_bCustomizeHovered = FALSE;
+                if (g_bCustomizeCapturing && ReleaseCapture())
+                {
+                    g_bCustomizeCapturing = FALSE;
+                }
+            }
+            return 0;
+        case WM_LBUTTONDOWN:
+            g_bCustomizeDown = TRUE;
+            break;
+        /* Hide tray overflow and open Notification Area Icons CPL */
+        case WM_LBUTTONUP:
+            if (g_bCustomizeDown)
+            {
+                g_bCustomizeDown = FALSE;
+                HWND hChevron = GetTrayChevron();
+                if (hChevron)
+                {
+                    SendMessageW(hChevron, WM_LBUTTONUP, NULL, NULL);
+                }
+                ShellExecuteW(
+                    NULL,
+                    L"open",
+                    L"explorer.exe",
+                    L"shell:::{26EE0668-A00A-44D7-9371-BEB064C98683}\\0\\::{05D7B0F4-2121-4EFF-BF6B-ED3F69B894D9}",
+                    NULL,
+                    SW_SHOWNORMAL
+                );
+            }
+            break;
+        case WM_CAPTURECHANGED:
+            g_bCustomizeCapturing = FALSE;
+            g_bCustomizeHovered = FALSE;
+            return 0;
+        case WM_NCDESTROY:
+            WindhawkUtils::RemoveWindowSubclassFromAnyThread(
```

Comment:
This is done automatically by the API. You can see the code by searching for `WM_NCDESTROY` in `windhawk_utils.h`.